com.unity.render-pipelines.core/Editor/Settings/RenderPipelineGlobalSettingsProvider.cs

```diff
@@ -0,0 +1,154 @@
+        /// </summary>
+        protected abstract void Ensure();
+
+        /// <summary>
+        /// Creates a new <see cref="RenderPipelineGlobalSettings"/> asset
+        /// </summary>
+        protected abstract void Create(bool useProjectSettingsFolder, bool activateAsset);
+
+        /// <summary>
+        /// Clones the <see cref="RenderPipelineGlobalSettings"/> asset
+        /// </summary>
+        protected abstract void Clone(RenderPipelineGlobalSettings src, bool activateAsset);
+
+        /// <summary>
+        /// Method called to render the IMGUI of the settings provider
+        /// </summary>
+        /// <param name="searchContext">The search content</param>
+        public override void OnGUI(string searchContext)
+        {
+            using (new SettingsProviderGUIScope())
+            {
+                if (renderPipelineSettings == null)
+                {
+                    CoreEditorUtils.DrawFixMeBox(string.Format(Styles.warningGlobalSettingsMissing, ObjectNames.NicifyVariableName(typeof(TGlobalSettings).Name)), () => Ensure());
+                }
+                else
+                {
+                    DrawAssetSelection();
+
+                    if (RenderPipelineManager.currentPipeline != null && !(RenderPipelineManager.currentPipeline is TRenderPipeline))
+                    {
+                        EditorGUILayout.HelpBox(string.Format(Styles.warningSRPNotActive, ObjectNames.NicifyVariableName(RenderPipelineManager.currentPipeline.GetType().Name)), MessageType.Warning);
+                    }
+
+                    if (m_Editor == null)
+                        m_Editor = Editor.CreateEditor(renderPipelineSettings);
+
+                    m_Editor.OnInspectorGUI();
+                }
+            }
+
+            base.OnGUI(searchContext);
+        }
+
+        void DrawAssetSelection()
+        {
+            using (new EditorGUILayout.HorizontalScope())
+            {
+                var newSettings = (TGlobalSettings)EditorGUILayout.ObjectField(renderPipelineSettings, typeof(TGlobalSettings), false);
+
+                if (renderPipelineSettings != newSettings)
+                {
+                    if (newSettings != null)
+                        GraphicsSettings.RegisterRenderPipelineSettings<TRenderPipeline>(newSettings);
+                    else
+                    {
+                        if (EditorUtility.DisplayDialog($"Invalid {ObjectNames.NicifyVariableName(typeof(TGlobalSettings).Name)}", Styles.settingNullRPSettings, "Yes", "No"))
+                            GraphicsSettings.UnregisterRenderPipelineSettings<TRenderPipeline>();
+                    }
+
+                    if (renderPipelineSettings != null && !renderPipelineSettings.Equals(null))
+                        EditorUtility.SetDirty(renderPipelineSettings);
+                }
+
+                if (GUILayout.Button(Styles.newAssetButtonLabel, GUILayout.Width(45), GUILayout.Height(18)))
```

Comment:
Extract these magic number to nicely named variables, since you are using them again further down